controllers/extendeddaemonsetreplicaset/strategy/canary.go

```diff
@@ -194,7 +196,11 @@ func manageCanaryPodFailures(pods []*v1.Pod, params *Parameters, result *Result,
 				"MaxRestarts", autoFailMaxRestarts,
 				"Reason", highRestartReason,
 			)
-		} else if !result.IsPaused && autoPauseEnabled {
+		case result.IsUnpaused:
```

Comment:
@clamoriniere this is also a key change made since your review. afaict we didn't have a way of updating the ERS `paused` condition to `false`, (so once `paused: true` we were in a positive cycle) so making it explicit here. i believe for `failed` it doesn't matter since the ERS should drop out